src/NServiceBus.Testing.Tests/Handler/HandlerTests.cs

```diff
@@ -61,6 +61,28 @@ public void SendMessageWithMultiIncomingHeaders()
         }
 
         [Test]
+        public void ShouldBeAbleToConfigureMessageHandlerContext()
+        {
+            var messageId = Guid.NewGuid().ToString();
+            var replyToAddress = "0118 999 881 999 119 725 3";
+            var handler = new ContextAccessingHandler();
+            TestableMessageHandlerContext contextInstance = null;
+
+            Test.Handler(handler)
+                .ConfigureHandlerContext(c =>
```

Comment:
I don't think that works with the current implementation. The handlers needs a `TestingContext` which derives from `TestableMessageHandlerContext`. This context contains the expectations and therefore the user can't just provide a custom instance (that's also why it's an action and not a function).

Even if we remove the expectations from the context (which I think we should), there are still the timeout messages in there which are specific to `Handler` and `Saga` tests, so I don't think it's worth the effort to work around that right now. This way at least both sagas and handler use the same API.